File: cs/Iris/ObjectType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Iris
{
    public class ObjectType
    {
        public static readonly ObjectType intType = new ObjectType("Int");
        public static readonly ObjectType stringType = new ObjectType("String");
        public static readonly ObjectType charType = new ObjectType("Char");
        public static readonly ObjectType booleanType = new ObjectType("Boolean");
        //public static readonly ObjectType undefinedType = new ObjectType("Undefined");
        public static readonly ObjectType undefinedType = new ObjectType(string.Empty);
        public static readonly ObjectType characterType = new ObjectType("Character");
        public static readonly ObjectType imageType = new ObjectType("Image");
        public static readonly ObjectType soundType = new ObjectType("Sound");

        public string name;

        public ObjectType()
        {

        }

        public ObjectType(string typeName)
        {
            name = typeName;
        }

        public static bool operator ==(ObjectType o1, ObjectType o2)
        {
            if (o1.name == string.Empty)
            {
                return true;
            }
            if (o2.name == string.Empty)
            {
                return true;
            }

            return o1.name.Equals(o2.name);
        }

        public static bool operator !=(ObjectType o1, ObjectType o2)
        {
            return !(o1 == o2);
        }

    }
}

File: java/Iris/src/com/rokuan/iris/types/ObjectType.java
package com.rokuan.iris.types;

public class ObjectType
{
	public static final ObjectType intType = new ObjectType("Int");
	public static final ObjectType stringType = new ObjectType("String");
	public static final ObjectType charType = new ObjectType("Char");
	public static final ObjectType booleanType = new ObjectType("Boolean");
	//public static final ObjectType undefinedType = new ObjectType("Undefined");
	public static final ObjectType undefinedType = new ObjectType("");
	public static final ObjectType characterType = new ObjectType("Character");
	public static final ObjectType imageType = new ObjectType("Image");
	public static final ObjectType soundType = new ObjectType("Sound");

	public String name;

	public ObjectType()
	{

	}

	public ObjectType(String typeName)
	{
		name = typeName;
	}

	@Override
	public boolean equals(Object o)
	{
		if(o == null){
			return false;
		}
		
		if(this == o){
			return true;
		}
		
		ObjectType ot = (ObjectType)o;

		if (this.name.isEmpty())
		{
			return true;
		}
		if (ot.name.isEmpty())
		{
			return true;
		}

		return this.name.equals(ot.name);
	}
}
